internal/js_parser/js_parser.go

```diff
@@ -7999,17 +7999,30 @@ func (p *parser) visitAndAppendStmt(stmts []js_ast.Stmt, stmt js_ast.Stmt) []js_
 				hasNumericValue = false
 				switch e := value.Value.Data.(type) {
 				case *js_ast.ENumber:
-					valuesSoFar[name] = e.Value
+					valuesSoFar[name] = e
 					hasNumericValue = true
 					nextNumericValue = e.Value + 1
 				case *js_ast.EString:
 					hasStringValue = true
+					if s.IsConst {
+						valuesSoFar[name] = e
+					}
+				default:
+					if s.IsConst {
+						r := js_lexer.RangeOfIdentifier(p.source, value.Value.Loc)
+						p.log.AddRangeError(&p.source, r, "const enum member initializers can only contain literal values and other computed enum values")
```

Comment:
If the const enum is exported, then the enum will still be emitted, given current logic. Thus, despite the error, the code will work. Do you think it is worth it to leave the error in, and not expose internals (that const enums are still emitted when they are exported)?

The same goes for the _Enum member must have initializer_ error.